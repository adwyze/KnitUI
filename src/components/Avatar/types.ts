import { SyntheticEvent } from 'react'

export interface IAvatarProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  size?: string
  onClick?: (event?: SyntheticEvent) => void
  picture?: string
  name?: string
  disabled?: boolean
}

export interface IAvatarBaseProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isPicture: boolean
  isName: boolean
  isIcon: boolean
  size: string
  disabled: boolean
}