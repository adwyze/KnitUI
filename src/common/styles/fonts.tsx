const InterThin = require("../assets/fonts/Inter-Thin-BETA.woff")
const InterLight = require("../assets/fonts/Inter-Light-BETA.woff")
const InterRegular = require("../assets/fonts/Inter-Regular.woff")
const InterMedium = require("../assets/fonts/Inter-Medium.woff")
const InterSemiBold = require("../assets/fonts/Inter-SemiBold.woff")
const InterBold = require("../assets/fonts/Inter-Bold.woff")
const InterBlack = require("../assets/fonts/Inter-Black.woff")

export const fontsDefinition = `
  @font-face {
    font-family: "InterThin";
    src: url(${InterThin}) format('woff'); /*URL to font*/
  }

  @font-face {
    font-family: "InterLight";
    src: url(${InterLight}) format('woff');
  }

  @font-face {
    font-family: "InterRegular";
    src: url(${InterRegular}) format('woff');
  }

  @font-face {
    font-family: "InterMedium";
    src: url(${InterMedium}) format('woff');
  }

  @font-face {
    font-family: "InterSemiBold";
    src: url(${InterSemiBold}) format('woff');
  }

  @font-face {
    font-family: "InterBold";
    src: url(${InterBold}) format('woff');
  }

  @font-face {
    font-family: "InterBlack";
    src: url(${InterBlack}) format('woff');
  }
`

export const fontsSource = {
  InterThin,
  InterLight,
  InterRegular,
  InterMedium,
  InterBold,
  InterBlack,
  InterSemiBold,
}
