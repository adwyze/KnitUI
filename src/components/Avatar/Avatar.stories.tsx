import React from "react"
import { storiesOf } from "@storybook/react"
import Avatar from "./"
import {
  withKnobs,
  text,
  boolean,
  optionsKnob as options,
} from "@storybook/addon-knobs"
import { action } from "@storybook/addon-actions"
const Readme = require("./README.md")

const stories = storiesOf("Avatar", module)
stories.addDecorator(withKnobs)

const defaultProps = {
  size: text("Size", "24px"),
  disabled: boolean("Disabled", false),
  onClick: action("button-click"),
}

const additionalProps = {
  picture: text("Picture", "https://i.ibb.co/DwVLSqF/user-profile.jpg"),
  name: text("Name", "Jane Doe"),
}

stories
  .addParameters({
    readme: {
      // Show readme before story
      codeTheme: "shades-of-purple",
      // Show readme at the addons panel
      sidebar: Readme,
    },
  })
  .add("Picture avatar", () => (
    <Avatar
      {...defaultProps}
      picture={additionalProps.picture}
      name={additionalProps.name}
    />
  ))
  .add("Name avatar", () => (
    <Avatar {...defaultProps} name={additionalProps.name} />
  ))
  .add("Unknown user avatar", () => <Avatar {...defaultProps} />)
  .add("Different size picture avatar", () => (
    <Avatar {...defaultProps} size="100px" {...additionalProps} />
  ))
  .add("Different size name avatar", () => (
    <Avatar {...defaultProps} size="100px" name={additionalProps.name} />
  ))
  .add("Different size icon avatar", () => (
    <Avatar {...defaultProps} size="100px" />
  ))
  .add("Disabled picture avatar", () => (
    <Avatar {...defaultProps} disabled {...additionalProps} />
  ))
  .add("Disabled name avatar", () => (
    <Avatar {...defaultProps} disabled name={additionalProps.name} />
  ))
  .add("Disabled icon avatar", () => <Avatar {...defaultProps} disabled />)
