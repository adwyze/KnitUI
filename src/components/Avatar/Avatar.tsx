import React, { FC, SyntheticEvent } from "react"
import styled from "styled-components"
import { IStyled } from "common/types"
import { oAccountCircle as UserProfileIcon } from "../Icon/Icons"

interface AvatarProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  size?: string
  onClick?: (event?: SyntheticEvent) => void
  picture?: string
  name?: string
  disabled: boolean
}

interface AvatarBaseProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isPicture: boolean
  isName: boolean
  isIcon: boolean
  size: string
  disabled: boolean
}

const getThemeColor = (props, color) => {
  const {
    theme: {
      knitui: { chromaPalette },
    },
  } = props
  return chromaPalette[color].hex()
}

const getBoxShadowStyle = props => {
  const {
    theme: {
      knitui: { shades },
    },
    customProps: { disabled },
  } = props

  return disabled ? "none" : `0px 0px 2px ${shades.blue50.hex()}`
}

const getFontSize = props => {
  const {
    customProps: { size },
  } = props
  return `calc(1.4 * ${size} / 2.4)`
}

const getLineHeight = props => {
  const {
    customProps: { size },
  } = props
  return `calc(2.0 * ${size} / 2.4)`
}

const getBackgroundColor = (props, state) => {
  const { isPicture, isName, isIcon, disabled } = props.customProps
  if (isPicture || isIcon) {
    return "transparent"
  } else if (isName) {
    if (disabled || state === "default") {
      return getThemeColor(props, "Neutral10")
    } else if (state === "hover" || state === "focus") {
      return getThemeColor(props, "Neutral20")
    } else {
      return getThemeColor(props, "Neutral10")
    }
  }
}

const getBorderStyle = (props, state) => {
  const { isPicture, isName, disabled } = props.customProps

  const focusColor = getThemeColor(props, "Azure80")
  const hoverColor = getThemeColor(props, "Neutral20")
  const defaultColor = getThemeColor(props, "Neutral10")

  if (state === "focus" && !disabled) {
    return `1px solid ${focusColor}`
  }
  if (isPicture || isName) {
    if (state === "default" || disabled) {
      return `1px solid ${defaultColor}`
    } else if (state === "hover") {
      return `1px solid ${hoverColor}`
    }
  }
  /* if(isIcon) with (state === "default" || state === "hover" || disabled), focus is already taken care of,
   * so directly returning transparent style in all other case of isIcon
   * because of the ts-lint error `not all paths return` to make this condition default
   */
  return `1px solid transparent`
}

const getMarginStyle = props => {
  const { isPicture, isIcon } = props.customProps
  return isIcon || isPicture ? "-1px" : "0"
}

const getFillColor = (props, state) => {
  const { isPicture, isName, isIcon, disabled } = props.customProps

  if (isPicture || isName) {
    return "transparent"
  } else if (isIcon) {
    if (state === "default" || disabled) {
      return getThemeColor(props, "Neutral10")
    } else if (state === "hover" || state === "focus") {
      return getThemeColor(props, "Neutral20")
    }
  }
}

const getOpacity = props => {
  const { isIcon } = props.customProps

  return isIcon ? 1 : 0.4
}

const AvatarBase = styled.button<IStyled<AvatarBaseProps>>`
  width: ${({ customProps }) => customProps.size};
  height: ${({ customProps }) => customProps.size};

  overflow: hidden;
  border-radius: 50%;
  outline: none;
  cursor: pointer;

  /* Style for Name Avatar */
  color: ${props => getThemeColor(props, "Neutral90")};
  font-size: ${props => getFontSize(props)};
  line-height: ${props => getLineHeight(props)};

  /* Below Basic Property & corresponding function calls which will play role in :
   * all three types of Avatar -
   *  - isPicture
   *  - isName
   *  - isIcon
   * along with thier 3 state & disabled attribute -
   *  - default
   *  - hover
   *  - focus/active
   */
  background: ${props => getBackgroundColor(props, "default")};
  border: ${props => getBorderStyle(props, "default")};
  img {
    width: ${({ customProps }) => customProps.size};
    height: ${({ customProps }) => customProps.size};
    margin: ${props => getMarginStyle(props)};
    user-drag: none;
    user-select: none;
    transition: transform 120ms;
  }

  svg {
    margin: ${props => getMarginStyle(props)};
    path {
      fill: ${props => getFillColor(props, "default")};
    }
  }

  :hover {
    img {
      transform: scale(1.05);
    }
    background: ${props => getBackgroundColor(props, "hover")};
    border: ${props => getBorderStyle(props, "hover")};
    svg path {
      fill: ${props => getFillColor(props, "hover")};
    }
  }
  :focus,
  :active {
    background: ${props => getBackgroundColor(props, "focus")};
    border: ${props => getBorderStyle(props, "focus")};
    svg path {
      fill: ${props => getFillColor(props, "focus")};
    }
    box-shadow: ${props => getBoxShadowStyle(props)};
  }

  :disabled {
    img {
      transform: none;
    }
    opacity: ${props => getOpacity(props)};
    cursor: not-allowed;
  }
`

const Avatar: FC<AvatarProps> = props => {
  const { picture, size = "24px", name, disabled = false, ...rest } = props

  const isPicture = !!picture
  const isName = !!name
  const isIcon = !(isPicture || isName)
  const avatarTitle = isName ? name : isPicture ? "" : "Unknown User"
  const customProps = {
    isPicture,
    isName,
    isIcon,
    size,
    disabled,
  }

  return (
    <AvatarBase
      data-testid="avatar"
      customProps={customProps}
      disabled={disabled}
      title={avatarTitle}
      {...rest}>
      {(isPicture && <img src={picture} />) ||
        (isName && name!.toUpperCase()[0]) ||
        (isIcon && <UserProfileIcon width={size} height={size} />)}
    </AvatarBase>
  )
}

export default Avatar
