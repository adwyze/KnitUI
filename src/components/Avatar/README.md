# Avatar

Avatar are useful to show information of user like user-profile, user-name. (it act as button, so according to usecase, valid to pass onClick or any html button attributes.)

## Usage

```javascript
import { Button } from "KnitUI"

<Avatar
  picture="www.url.com/image.jpg"
  name="username"
  size="30px"
  disabled={false}
  {...anyHtmlButtonProps}
/>)
```

### Props

| Prop name | Type       | Optional | Default     | Description                                                                                                       |
| --------- | ---------- | -------- | ----------- | ----------------------------------------------------------------------------------------------------------------- |
| picture   | string     | Yes      | `undefined` | url of the profile picture or any valid component for react img src attribute                                     |
| name      | string     | Yes      | `undefined` | user name which first letter will be shown, if no picture url is passed or falsy picture value                    |
| size      | string     | Yes      | `24px`      | Physical area occupied on the screen, any css valid length parameter, (resize the font accordingly if only in px) |
| disabled  | boolean    | Yes      | `false`     | Whether the Avatar should be interactive or not                                                                   |
| onClick   | Function   | Yes      | None        | A click handler to be executed on click of the button. Will receive the `event` as an argument                    |
| className | string     | Yes      | None        | CSS class name to be added                                                                                        |
| style     | CSS Object | Yes      | None        | CSS style to be added                                                                                             |
