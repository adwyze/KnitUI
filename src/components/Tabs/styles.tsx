import { RefObject } from "react"
import styled, { createGlobalStyle } from "styled-components"
import { Button, Icon } from ".."
import { getThemeColor, getOSName } from "../../common/_utils"

/* The tab being dragged is appended to the body, hence it's styles need to be at global scope*/

export const GlobalStyle = createGlobalStyle`
.knitui-tabs-helper:not(#knit-active-tab) {
  background: ${props => getThemeColor(props, "Beige10")};
  
  &:before {
    content: "";
    position: absolute;
    left: 0;
    top: 50%;
    transform: translateY(-50%);
    height: 1.4rem;
    width: 1px;
    background: ${props => getThemeColor(props, "Neutral40")};
  }
}

.knitui-tabs-helper {
  white-space: nowrap;
}
`

export const ButtonWrapper = styled(Button)`
  border-radius: 4px 4px 0 0;
  :hover,
  :focus,
  :active {
    background: #efe9dc;
  }
`

export const TabsWrapper = styled.div<{ hideTabContent: boolean }>`
  height: ${({ hideTabContent }) => (hideTabContent ? "auto" : "100%")};
  width: 100%;
`

export const TabsPanelWrapper = styled.div`
  display: flex;
  background: ${props => getThemeColor(props, "Beige10")};
`

const getActiveTabHeight = props => {
  return "100%"

  /**
   * Idea was to assign height as much as active tab but during drag tab
   * height has extra 2px which causing recursive increase of OverflowWrapper's height.
   * Have to find work-around for this.
   */
  return props.activeTabRef.current
    ? `${props.activeTabRef.current.getBoundingClientRect().height}px`
    : "100%"
}

export const OverflowWrapper = styled.div<{
  activeTabRef: RefObject<HTMLDivElement>
}>`
  /**
   * Height needed to taken from ref because in some cases like in MacOS Mojave 
   * irrespective of browser & it's version, padding-bottom still visible.
   */
  height: ${props => getActiveTabHeight(props)};
  overflow: hidden;
  position: relative;
  display: flex;
`

export const BlurElement = styled.div<{ dir: string; visible: boolean }>`
  width: 80px;
  height: 100%;
  position: absolute;
  background: linear-gradient(
    to ${props => (props.dir === "left" ? "right" : "left")},
    ${props => getThemeColor(props, "Beige10")}FF,
    ${props => getThemeColor(props, "Beige10")}00
  );
  z-index: 100;
  visibility: ${props => (props.visible ? "visible" : "hidden")};
  pointer-events: none;
`

export const IconWrapper = styled.div<{ visible: boolean }>`
  display: flex;
  justify-content: center;
  align-items: center;
  transition: visibility 250ms ease, width 150ms linear;
  visibility: ${props => (props.visible ? "visible" : "hidden")};
  width: ${props => (props.visible ? "30px" : "0")};
  cursor: pointer;
  z-index: 100;

  :hover,
  :focus,
  :active {
    background: #efe9dc;
  }
`

export const TabPanel = styled.button<{
  active?: boolean
  dragHandle?: boolean
}>`
  display: flex;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  word-wrap: none;

  background: ${props =>
    props.active ? getThemeColor(props, "Neutral0") : "none"};
  color: ${props =>
    props.active
      ? getThemeColor(props, "Neutral90")
      : getThemeColor(props, "Neutral50")};
  border: 2px solid transparent;
  border-top: ${props =>
    props.active ? "2px solid #D8C9A7" : "2px solid transparent"};
  padding: 4px 14px;
  border-radius: 4px 4px 0px 0px;
  min-width: 80px;
  line-height: 20px;
  font-size: 14px;
  cursor: ${props => (props.dragHandle && props.active ? "auto" : "pointer")};

  /* To prevent focus outline overlapping with divider */
  margin-right: ${props => (props.active ? "0" : "1px")};

  :hover {
    background: ${props => (props.active ? "#FFFFFF" : "#efe9dc")};
  }

  :active,
  :focus {
    z-index: 101 !important;
    outline-color: ${props => getThemeColor(props, "Azure80")};
    border-color: ${props => getThemeColor(props, "Azure80")} auto 1px;
  }
`

export const TabContentWrapper = styled.div`
  width: 100%;
`

export const OverflowContainer = styled.div`
  display: flex;
  align-items: center;
  white-space: nowrap;
  width: 100%;
  overflow: scroll hidden;
  padding-bottom: ${getOSName() !== "MacOS" ? "0px" : "17px"};
  margin-bottom: -17px;
  box-sizing: content-box;
`

export const DragIcon = styled(Icon)`
  cursor: grab;
  & svg {
    fill: ${props => getThemeColor(props, "Neutral45")};
  }
  margin-right: 0.4rem;
  & svg :hover,
  & svg :active,
  & svg :focus {
    fill: #000000;
  }
`

export const VerticalBar = styled.div`
  height: 14px;
  width: 0px;
  position: absolute;
  border-right: 1px solid #cccccc;
  top: 50%;
  right: 0;
  transform: translateY(-50%);
  pointer-events: none;
`
