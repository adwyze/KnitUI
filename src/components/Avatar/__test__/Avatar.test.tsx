import React from "react"
import Avatar from "../"
import { render, cleanup, fireEvent } from "react-testing-library"
import { ThemeProvider } from "../../../common/styles"
import "jest-styled-components"
import "jest-dom/extend-expect"

afterEach(cleanup)

const defaultProps = {
  size: "24px",
  disabled: false,
  onClick: () => {},
}

const additionalProps = {
  picture: "https://i.ibb.co/DwVLSqF/user-profile.jpg",
  name: "Jane Doe",
}

describe("Avatar Snapshots", () => {
  it("Picture avatar", () => {
    const { asFragment } = render(
      <Avatar {...defaultProps} {...additionalProps} />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Name avatar", () => {
    const { asFragment } = render(
      <Avatar {...defaultProps} name={additionalProps.name} />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Unknown user avatar", () => {
    const { asFragment } = render(<Avatar {...defaultProps} />, {
      wrapper: ThemeProvider,
    })
    expect(asFragment()).toMatchSnapshot()
  })
  it("Different size picture avatar", () => {
    const { asFragment } = render(
      <Avatar {...defaultProps} size="100px" {...additionalProps} />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Different size name avatar", () => {
    const { asFragment } = render(
      <Avatar {...defaultProps} size="100px" name={additionalProps.name} />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Different size icon avatar", () => {
    const { asFragment } = render(<Avatar {...defaultProps} size="100px" />, {
      wrapper: ThemeProvider,
    })
    expect(asFragment()).toMatchSnapshot()
  })
  it("Disabled picture avatar", () => {
    const { asFragment } = render(
      <Avatar {...defaultProps} size="100px" disabled {...additionalProps} />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Disabled name avatar", () => {
    const { asFragment } = render(
      <Avatar
        {...defaultProps}
        size="100px"
        disabled
        name={additionalProps.name}
      />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("Disabled icon avatar", () => {
    const { asFragment } = render(
      <Avatar {...defaultProps} disabled size="100px" />,
      { wrapper: ThemeProvider }
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("Avatar onClick Functionality", () => {
  const spyOnClick = jest.spyOn(defaultProps, "onClick")

  afterEach(() => {
    spyOnClick.mockReset()
  })
  it("passed onClick function get fired when clicked", () => {
    const { getByTestId } = render(
      <Avatar {...defaultProps} size="100px" name={additionalProps.name} />,
      { wrapper: ThemeProvider }
    )

    fireEvent.click(getByTestId("avatar"))
    expect(spyOnClick).toBeCalledTimes(1)
  })

  it("passed onClick function does not get fired when clicked on Disabled Avatar", () => {
    const { getByTestId } = render(
      <Avatar
        {...{ ...defaultProps, disabled: true }}
        size="100px"
        name={additionalProps.name}
      />,
      { wrapper: ThemeProvider }
    )

    fireEvent.click(getByTestId("avatar"))
    expect(spyOnClick).toBeCalledTimes(0)
  })
})
