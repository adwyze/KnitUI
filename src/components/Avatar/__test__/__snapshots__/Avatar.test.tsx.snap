// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Snapshots Different size icon avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 5.833333333333334rem;
  line-height: 8.333333333333334rem;
  background: transparent;
  border: 1px solid transparent;
}

.c0 img {
  width: 100px;
  height: 100px;
  margin: -1px;
}

.c0 svg {
  margin: -1px;
}

.c0 svg path {
  fill: #f2f2f2;
}

.c0:hover {
  background: transparent;
  border: 1px solid transparent;
}

.c0:hover svg path {
  fill: #e6e6e6;
}

.c0:focus,
.c0:active {
  background: transparent;
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
}

.c0:focus svg path,
.c0:active svg path {
  fill: #e6e6e6;
}

.c0:disabled {
  opacity: 1;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    title="Unknown User"
  >
    <svg
      fill="#000"
      height="100px"
      viewBox="0 0 24 24"
      width="100px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM7.07 18.28C7.5 17.38 10.12 16.5 12 16.5C13.88 16.5 16.51 17.38 16.93 18.28C15.57 19.36 13.86 20 12 20C10.14 20 8.43 19.36 7.07 18.28ZM18.36 16.83C16.93 15.09 13.46 14.5 12 14.5C10.54 14.5 7.07 15.09 5.64 16.83C4.62 15.49 4 13.82 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 13.82 19.38 15.49 18.36 16.83ZM12 6C10.06 6 8.5 7.56 8.5 9.5C8.5 11.44 10.06 13 12 13C13.94 13 15.5 11.44 15.5 9.5C15.5 7.56 13.94 6 12 6ZM12 11C11.17 11 10.5 10.33 10.5 9.5C10.5 8.67 11.17 8 12 8C12.83 8 13.5 8.67 13.5 9.5C13.5 10.33 12.83 11 12 11Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Different size name avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 5.833333333333334rem;
  line-height: 8.333333333333334rem;
  background: #f2f2f2;
  border: none;
}

.c0 img {
  width: 100px;
  height: 100px;
  margin: 0;
}

.c0 svg {
  margin: 0;
}

.c0 svg path {
  fill: transparent;
}

.c0:hover {
  background: #e6e6e6;
  border: none;
}

.c0:hover svg path {
  fill: transparent;
}

.c0:focus,
.c0:active {
  background: #e6e6e6;
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
}

.c0:focus svg path,
.c0:active svg path {
  fill: transparent;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    title="Jane Doe"
  >
    J
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Different size picture avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 5.833333333333334rem;
  line-height: 8.333333333333334rem;
  background: transparent;
  border: 1px solid #f2f2f2;
}

.c0 img {
  width: 100px;
  height: 100px;
  margin: -1px;
}

.c0 svg {
  margin: -1px;
}

.c0 svg path {
  fill: transparent;
}

.c0:hover {
  background: transparent;
  border: 1px solid #e6e6e6;
}

.c0:hover svg path {
  fill: transparent;
}

.c0:focus,
.c0:active {
  background: transparent;
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
}

.c0:focus svg path,
.c0:active svg path {
  fill: transparent;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    title="Jane Doe"
  >
    <img
      src="https://i.ibb.co/DwVLSqF/user-profile.jpg"
    />
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Disabled icon avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 5.833333333333334rem;
  line-height: 8.333333333333334rem;
  background: transparent;
  border: 1px solid transparent;
}

.c0 img {
  width: 100px;
  height: 100px;
  margin: -1px;
}

.c0 svg {
  margin: -1px;
}

.c0 svg path {
  fill: #f2f2f2;
}

.c0:hover {
  background: transparent;
  border: 1px solid transparent;
}

.c0:hover svg path {
  fill: #f2f2f2;
}

.c0:focus,
.c0:active {
  background: transparent;
  border: 1px solid transparent;
  box-shadow: none;
}

.c0:focus svg path,
.c0:active svg path {
  fill: #f2f2f2;
}

.c0:disabled {
  opacity: 1;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    disabled=""
    title="Unknown User"
  >
    <svg
      fill="#000"
      height="100px"
      viewBox="0 0 24 24"
      width="100px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM7.07 18.28C7.5 17.38 10.12 16.5 12 16.5C13.88 16.5 16.51 17.38 16.93 18.28C15.57 19.36 13.86 20 12 20C10.14 20 8.43 19.36 7.07 18.28ZM18.36 16.83C16.93 15.09 13.46 14.5 12 14.5C10.54 14.5 7.07 15.09 5.64 16.83C4.62 15.49 4 13.82 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 13.82 19.38 15.49 18.36 16.83ZM12 6C10.06 6 8.5 7.56 8.5 9.5C8.5 11.44 10.06 13 12 13C13.94 13 15.5 11.44 15.5 9.5C15.5 7.56 13.94 6 12 6ZM12 11C11.17 11 10.5 10.33 10.5 9.5C10.5 8.67 11.17 8 12 8C12.83 8 13.5 8.67 13.5 9.5C13.5 10.33 12.83 11 12 11Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Disabled name avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 5.833333333333334rem;
  line-height: 8.333333333333334rem;
  background: #f2f2f2;
  border: none;
}

.c0 img {
  width: 100px;
  height: 100px;
  margin: 0;
}

.c0 svg {
  margin: 0;
}

.c0 svg path {
  fill: transparent;
}

.c0:hover {
  background: #f2f2f2;
  border: none;
}

.c0:hover svg path {
  fill: transparent;
}

.c0:focus,
.c0:active {
  background: #f2f2f2;
  border: none;
  box-shadow: none;
}

.c0:focus svg path,
.c0:active svg path {
  fill: transparent;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    disabled=""
    title="Jane Doe"
  >
    J
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Disabled picture avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 100px;
  height: 100px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 5.833333333333334rem;
  line-height: 8.333333333333334rem;
  background: transparent;
  border: 1px solid #f2f2f2;
}

.c0 img {
  width: 100px;
  height: 100px;
  margin: -1px;
}

.c0 svg {
  margin: -1px;
}

.c0 svg path {
  fill: transparent;
}

.c0:hover {
  background: transparent;
  border: 1px solid #f2f2f2;
}

.c0:hover svg path {
  fill: transparent;
}

.c0:focus,
.c0:active {
  background: transparent;
  border: 1px solid #f2f2f2;
  box-shadow: none;
}

.c0:focus svg path,
.c0:active svg path {
  fill: transparent;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    disabled=""
    title="Jane Doe"
  >
    <img
      src="https://i.ibb.co/DwVLSqF/user-profile.jpg"
    />
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Name avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 1.4000000000000001rem;
  line-height: 2.0000000000000004rem;
  background: #f2f2f2;
  border: none;
}

.c0 img {
  width: 24px;
  height: 24px;
  margin: 0;
}

.c0 svg {
  margin: 0;
}

.c0 svg path {
  fill: transparent;
}

.c0:hover {
  background: #e6e6e6;
  border: none;
}

.c0:hover svg path {
  fill: transparent;
}

.c0:focus,
.c0:active {
  background: #e6e6e6;
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
}

.c0:focus svg path,
.c0:active svg path {
  fill: transparent;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    title="Jane Doe"
  >
    J
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Picture avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 1.4000000000000001rem;
  line-height: 2.0000000000000004rem;
  background: transparent;
  border: 1px solid #f2f2f2;
}

.c0 img {
  width: 24px;
  height: 24px;
  margin: -1px;
}

.c0 svg {
  margin: -1px;
}

.c0 svg path {
  fill: transparent;
}

.c0:hover {
  background: transparent;
  border: 1px solid #e6e6e6;
}

.c0:hover svg path {
  fill: transparent;
}

.c0:focus,
.c0:active {
  background: transparent;
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
}

.c0:focus svg path,
.c0:active svg path {
  fill: transparent;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    title="Jane Doe"
  >
    <img
      src="https://i.ibb.co/DwVLSqF/user-profile.jpg"
    />
  </button>
</DocumentFragment>
`;

exports[`Avatar Snapshots Unknown user avatar 1`] = `
<DocumentFragment>
  .c0 {
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  overflow: hidden;
  border-radius: 50%;
  outline: none;
  color: #1a1a1a;
  font-size: 1.4000000000000001rem;
  line-height: 2.0000000000000004rem;
  background: transparent;
  border: 1px solid transparent;
}

.c0 img {
  width: 24px;
  height: 24px;
  margin: -1px;
}

.c0 svg {
  margin: -1px;
}

.c0 svg path {
  fill: #f2f2f2;
}

.c0:hover {
  background: transparent;
  border: 1px solid transparent;
}

.c0:hover svg path {
  fill: #e6e6e6;
}

.c0:focus,
.c0:active {
  background: transparent;
  border: 1px solid #5ab2f2;
  box-shadow: 0px 0px 2px #0066ff;
}

.c0:focus svg path,
.c0:active svg path {
  fill: #e6e6e6;
}

.c0:disabled {
  opacity: 1;
  cursor: not-allowed;
}

<button
    class="c0"
    data-testid="avatar"
    title="Unknown User"
  >
    <svg
      fill="#000"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM7.07 18.28C7.5 17.38 10.12 16.5 12 16.5C13.88 16.5 16.51 17.38 16.93 18.28C15.57 19.36 13.86 20 12 20C10.14 20 8.43 19.36 7.07 18.28ZM18.36 16.83C16.93 15.09 13.46 14.5 12 14.5C10.54 14.5 7.07 15.09 5.64 16.83C4.62 15.49 4 13.82 4 12C4 7.59 7.59 4 12 4C16.41 4 20 7.59 20 12C20 13.82 19.38 15.49 18.36 16.83ZM12 6C10.06 6 8.5 7.56 8.5 9.5C8.5 11.44 10.06 13 12 13C13.94 13 15.5 11.44 15.5 9.5C15.5 7.56 13.94 6 12 6ZM12 11C11.17 11 10.5 10.33 10.5 9.5C10.5 8.67 11.17 8 12 8C12.83 8 13.5 8.67 13.5 9.5C13.5 10.33 12.83 11 12 11Z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
